wardrobe/kitties/src/lib.rs

```diff
@@ -500,28 +561,111 @@ impl TryFrom<&DynamicallyTypedData> for KittyData {
+                }
+                Ok(0)
+            }
+            Self::Breed => {
+                // Check that we are consuming at least one input.
+                ensure!(input_data.len() == 2, Self::Error::TwoParentsDoNotExist);
+
+                let mom = KittyData::try_from(&input_data[0])?;
+                let dad = KittyData::try_from(&input_data[1])?;
+                KittyHelpers::can_breed(&mom, &dad)?;
+                // Output must be Mom, Dad, and Child.
+                ensure!(output_data.len() == 3, Self::Error::NotEnoughFamilyMembers);
+                KittyHelpers::check_new_family(&mom, &dad, output_data)?;
+                Ok(0)
+            }
+            Self::UpdateKittyName => {
+                can_kitty_name_be_updated(input_data, output_data)?;
+                Ok(0)
+            }
+        }
+    }
+}
 
-        Ok(0)
+/// Checks:
+///     - Input and output are of kittyType.
+///     - Only the name is updated, and other basic properties are not modified.
+///     - The order between input and output must be the same.
+pub fn can_kitty_name_be_updated(
+    input_data: &[DynamicallyTypedData],
+    output_data: &[DynamicallyTypedData],
+) -> Result<TransactionPriority, ConstraintCheckerError> {
+    ensure!(
+        input_data.len() == output_data.len() && !input_data.is_empty(),
+        { ConstraintCheckerError::NumberOfInputOutputMismatch }
+    );
+
+    for (input, output) in input_data.iter().zip(output_data.iter()) {
+        let utxo_input_kitty = input
+            .clone()
+            .extract::<KittyData>()
+            .map_err(|_| ConstraintCheckerError::BadlyTyped)?;
+
+        let utxo_output_kitty = output
+            .clone()
+            .extract::<KittyData>()
+            .map_err(|_| ConstraintCheckerError::BadlyTyped)?;
+
+        check_kitty_name_update(&utxo_input_kitty, &utxo_output_kitty)?;
     }
+    Ok(0)
+}
+
+/// Checks:
+///     - This is a private function used by can_kitty_name_be_updated.
+///     - Only the name is updated, and other basic properties are not updated.
+///
+fn check_kitty_name_update(
+    original_kitty: &KittyData,
+    updated_kitty: &KittyData,
+) -> Result<TransactionPriority, ConstraintCheckerError> {
+    ensure!(
+        original_kitty.dna == updated_kitty.dna,
+        ConstraintCheckerError::DnaMismatchBetweenInputAndOutput
+    );
+    ensure!(
+        original_kitty != updated_kitty,
+        ConstraintCheckerError::KittyNameUnAltered
+    );
+    ensure!(
+        original_kitty.free_breedings == updated_kitty.free_breedings,
+        ConstraintCheckerError::FreeBreedingCannotBeUpdated
+    );
+    ensure!(
+        original_kitty.num_breedings == updated_kitty.num_breedings,
+        ConstraintCheckerError::NumOfBreedingCannotBeUpdated
+    );
+    ensure!(
+        original_kitty.parent == updated_kitty.parent,
+        ConstraintCheckerError::KittyGenderCannotBeUpdated
+    );
```

Comment:
@JoshOrndorff do you think we should encourage this approach of multiple errors and verbosity?

Reply:
I believe it is preferable to provide a more granular error message, specifying the reason for the issue. High-level errors like 'BasicPropertiesAltered' may not offer sufficient information to developers, especially when dealing with multiple properties, making it challenging to discern the exact nature of the problem.

Reply:
I don't have an opinion. I'm fine to try it out this way and see what is better for downstream devs and end users.